*: modify tests to work with only-full-group-by on

morgo opened the pull request:
<!--
Thank you for contributing to TiDB! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
-->

### What problem does this PR solve? <!--add issue link with summary if exists-->

This PR allows the test-suite to run with the compiled default sql-mode including `ONLY_FULL_GROUP_BY`.  It does not add any new tests (yet).

### What is changed and how it works?

The change is to explicitly set the sql mode in tests that require it.  A better long time fix might be to use the `ANY_VALUE` function (needs work: https://github.com/pingcap/tidb/issues/8161 ), or try running tests with only-full-group-by on/off.

### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

 - Unit test

Code changes

- Only test changes

Side effects

 - Only sets sql-mode to strict_trans_tables.  If other modes are later turned on by default, behavior might be strange.

Related changes

- Need to fix `ANY_VALUE`: https://github.com/pingcap/tidb/issues/8161
- Can change default to only-full-group-by as on: https://github.com/pingcap/parser/pull/11

morgo commented:
/run-all-tests

shenli commented:
@morgo This PR makes sure that the current test cases are compatible with the `ONLY_FULL_GROUP_BY` sql-mode, right?

morgo commented:
@shenli that is correct.  I modified parser/mysql/ to enable only-full-group-by by default, and fixed all the tests that broke.

The tests are now stable in that they can work with either on or off.
